File: practices/r2dbc-webflux-slice-test/src/main/java/com/r2dbc/repository/UserR2dbcRepository.java
package com.r2dbc.repository;

import com.r2dbc.common.repository.UserEntity;
import org.springframework.data.r2dbc.repository.R2dbcRepository;
import reactor.core.publisher.Flux;

public interface UserR2dbcRepository extends R2dbcRepository<UserEntity, Long> {
    Flux<UserEntity> findAllByNameStartsWith(String prefix);
}

File: practices/coroutine-r2dbc/src/main/kotlin/com/coroutiner2dbc/repository/UserR2dbcRepository.kt
package com.coroutiner2dbc.repository

import com.coroutiner2dbc.common.repository.UserEntity
import org.springframework.data.repository.kotlin.CoroutineSortingRepository

/**
 * packageName    : com.coroutiner2dbc.repository
 * fileName       : UserR2dbcRepository
 * author         : okdori
 * date           : 3/8/24
 * description    :
 */

interface UserR2dbcRepository : CoroutineSortingRepository<UserEntity, Long>
